src/js/wizards/Buy/0_Contact/index.jsx

```diff
@@ -22,21 +22,19 @@ class Contact extends Component {
     this.validate(props.username, props.statusContactCode);
     props.footer.enableNext();
     props.footer.onPageChange(() => {
+      props.signMessage(this.state.username, this.state.statusContactCode);
       props.setContactInfo({username: DOMPurify.sanitize(this.state.username), statusContactCode: DOMPurify.sanitize(this.state.statusContactCode)});
     });
   }
 
   componentDidMount() {
     if (this.props.profile && this.props.profile.username) {
       this.props.setContactInfo({username: DOMPurify.sanitize(this.props.profile.username), statusContactCode: DOMPurify.sanitize(this.props.profile.statusContactCode)});
-      // FIXME: infinite loop between this page and the next
-      setTimeout(() => {
-        this.props.wizard.next();
-      }, 500);
     } else {
       this.validate(this.props.username, this.props.statusContactCode);
-      this.setState({ready: true});
     }
+
+    this.setState({ready: true});
```

Comment:
Ready doesn't really achieve much now. 
If we still want a loading, we can just put an `if` in the render and show the loading only `if (!this.props.profile)`